packages/components/src/color-palette/stories/index.tsx

```diff
@@ -43,10 +43,20 @@ const Template: ComponentStory< typeof ColorPalette > = ( {
 	onChange,
 	...args
 } ) => {
-	const firstColor =
-		( args.colors as ColorObject[] )[ 0 ].color ||
-		( args.colors as PaletteObject[] )[ 0 ].colors[ 0 ].color;
-	const [ color, setColor ] = useState< string | undefined >( firstColor );
+	let initialColor;
+	if ( args.color?.length ) {
+		const asColorObject = ( args.colors as ColorObject[] )?.[ 0 ].color;
+		let asPaletteObject;
+
+		if ( ( args.colors as PaletteObject[] )[ 0 ].colors?.length ) {
+			asPaletteObject = ( args.colors as PaletteObject[] )[ 0 ]
+				.colors[ 0 ].color;
+		}
+
+		initialColor = asColorObject || asPaletteObject;
```

Comment:
I personally find it more readable if we just added some question marks to the original code — what do you think?

```tsx
	const initialColor =
		( args.colors as ColorObject[] )?.[ 0 ]?.color ||
		( args.colors as PaletteObject[] )?.[ 0 ]?.colors[ 0 ].color;
```

FWIW I'm also fine with removing this "initial color" selection logic altogether. It's not part of the default component logic, and it's probably not a common use case either.

Reply:
I'd rather remove it altogether, to be honest